Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Sliders/PinchSlider.cs

```diff
@@ -0,0 +1,198 @@
+﻿//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+//
+using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine.Events;
+using Microsoft.MixedReality.Toolkit.Input;
+using System;
+
+namespace Microsoft.MixedReality.Toolkit.UI
+{
+    /// <summary>
+    /// Slider with a thumb (a GameObject), endpoints are defined by two GameObjects
+    /// </summary>
+    public class PinchSlider : MonoBehaviour, IMixedRealityPointerHandler, IMixedRealityFocusHandler
+    {
+
+        #region Serialized Fields and Properties
+        [Tooltip("The gameObject that contains all slider children. Moves the sliders together.")]
+        [SerializeField]
+        private GameObject sliderRoot = null;
+
+        [Range(0, 1)]
+        [SerializeField]
+        private float sliderValue = 0.5f;
+        public float SliderValue
+        {
+            get { return sliderValue; }
+            set
+            {
+                var oldSliderValue = sliderValue;
+                sliderValue = value;
+                UpdateUI();
+                OnValueUpdated.Invoke(new SliderEventData(oldSliderValue, value, activePointer is IMixedRealityPointer));
+            }
+        }
+
+        [Header("Slider Track")]
+
+        [Tooltip("The axis the slider moves along")]
+        [SerializeField]
+        private SliderAxis sliderAxis = SliderAxis.XAxis;
+        [Serializable]
+        /// <summary>
+        /// The axis the slider moves along
+        /// </summary>
+        private enum SliderAxis
+        {
+            XAxis = 0,
+            YAxis,
+            ZAxis
+        }
+
+        [SerializeField]
+        [Tooltip("Where the slider track starts, in local coordinates")]
+        private Vector3 sliderStartPosition = new Vector3(-0.5f, 0, 0);
+        [SerializeField]
```

Comment:
Could we simplify slider start and end positions to just a radius value (implicitly along the axis) or do you think we need the flexibility?

Reply:
I would prefer to keep this because it will make it easier to use our line classes as slider paths in future implementations.

Reply:
Going to keep this, first to make line computation easier, and second because I like being able to specify start and end points that aren't always centered about the slider.

Reply:
@Railboy I'm changing internal representation of start and end to be a distance from center of slider along the slider axis so that if you change the slider axis, the start and end handles maintain their distance to the center instead of snapping to zero. 

It should still be easy to use line classes to draw the slider track, since you can still get and set the start / end positions of the slider track in world space.